### PR TITLE
GetMessage + MediatR

### PR DESCRIPTION
**Pull request by mbusk88**
## Description

First draft for GetMessage:
- Able to receive HTTP call from market operator.
- Checking for DataAvailable in DB.
- Sending DataAvailable UUIDs to sub domain.
- Ready to receive path to market operator-data in DB (but NOT used in code as of now).
- Getting market operator-data in DB and responding to query (using hardcoded path and self-constructed data).
-  Added MediatR with one Query: GetMessageQuery. 

## References

* #62 #63 #64 #65

### CODE REVIEW

File: source/Energinet.DataHub.PostOffice.Outbound/Functions/Dequeue.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public class Dequeue
             _documentStore = documentStore;
         }
 
-        [Function("Dequeue")]
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Outbound/Functions/Dequeue.cs:36]: Remember to undo or do not commit temporarily changes.

File: source/Energinet.DataHub.PostOffice.Outbound/Functions/GetMessage.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                var data = await _mediator.Send(getMessageQuery).ConfigureAwait(false);
+
+                var response = request.CreateResponse(HttpStatusCode.OK);
+                await response.WriteAsJsonAsync(data).ConfigureAwait(false);
+
+                return response;
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException("Something went wrong... Baaah!", e);
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Outbound/Functions/GetMessage.cs:61]: Use better error descriptions than Baaaah

File: source/Energinet.DataHub.PostOffice.Outbound/local.settings.sample.json [hunk 3]
```diff
@@ -1,11 +1,14 @@
 {
   "IsEncrypted": false,
   "Values": {
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Outbound/local.settings.sample.json:3]: Remove local settings from sample settings

File: source/Energinet.DataHub.PostOffice.Application/GetMessage/ICosmosService.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Energinet.DataHub.PostOffice.Application.GetMessage
+{
+    /// <summary>
+    /// Service to connect and retrieve data from Cosmos database
+    /// </summary>
+    public interface ICosmosService
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Application/GetMessage/ICosmosService.cs:23]: Still cosmos specific

File: source/Energinet.DataHub.PostOffice.Application/GetMessage/IBlobStorageService.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+// limitations under the License.
+
+using System.Threading.Tasks;
+
+namespace Energinet.DataHub.PostOffice.Application.GetMessage
+{
+    /// <summary>
+    /// Service to query Azure blob storage containing data to external actors
+    /// </summary>
+    public interface IBlobStorageService
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Application/GetMessage/IBlobStorageService.cs:22]: Remove Blob from name, maybe just call it : IStorageService --- GetStorageContentAsync  in method.

File: source/Energinet.DataHub.PostOffice.Application/GetMessage/GetMessageHandler.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+        public async Task<string> Handle(GetMessageQuery request, CancellationToken cancellationToken)
+        {
+            if (request is null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var uuids = await _cosmosService.GetDataAvailableUuidsAsync(request.Recipient).ConfigureAwait(false);
+
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Application/GetMessage/GetMessageHandler.cs:55]: Should we call it more like: RequestDataFromSubDomain ?  or RequestAvailableDataFromSubDomain ?
If the name contains ToMarketOperator, I read it as only the program-flow for market operators can use this method for data.

File: source/Energinet.DataHub.PostOffice.Application/GetMessage/GetMessageHandler.cs [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+
+namespace Energinet.DataHub.PostOffice.Application.GetMessage
+{
+    public class GetMessageHandler : IRequestHandler<GetMessageQuery, string>
+    {
+        private readonly string _queueName = "charges";
+        private readonly string _blobStorageFileName = "Test.txt";
```
Comment (xnetsrak) [inline source/Energinet.DataHub.PostOffice.Application/GetMessage/GetMessageHandler.cs:26]: For now leave it. But I dont know if the Handler should be responsible for choosing correct queue and filename

**Review by xnetsrak — CHANGES_REQUESTED** [review]
See comments. Some changes and heads-ups before merge.

File: source/Energinet.DataHub.PostOffice.Application/GetMessage/GetMessageHandler.cs (hunk 6 above)
Comment (mbusk88) [reply source/Energinet.DataHub.PostOffice.Application/GetMessage/GetMessageHandler.cs:55]: I think we should discuss this more. I leave it as it is for now.

**Review by xnetsrak — DISMISSED** [review]
I will approve, still some naming I disagree with, but we can talk about that later.

**Review by xnetsrak — APPROVED** [review]

**Pull request merged**